resource/resourceadapters/opener.go

```diff
@@ -13,23 +13,40 @@ import (
 	corestate "github.com/juju/juju/state"
 )
 
+// NewResourceOpener returns a new resource.Opener for the given unit.
+//
+// TODO(mjs): This is the entry point for a whole lot of untested shim
+// code in this package. At some point this should be sorted out.
+func NewResourceOpener(st *corestate.State, unitName string) (opener resource.Opener, err error) {
```

Comment:
Can you add a comment that the caller is responsible for closing state just be make it explicit.
I traced to code to be sure we weren't leaking connections, so would be nice to inform the reader the expectation.